Nodejs/Product/Nodejs/Debugger/NodeDebugProvider.cs

```diff
@@ -58,6 +77,61 @@ public void SetupDebugTargetInfo(ref VsDebugTargetInfo vsDebugTargetInfo, DebugL
             vsDebugTargetInfo.grfLaunch = (uint)__VSDBGLAUNCHFLAGS.DBGLAUNCH_StopDebuggingOnEnd;
         }
 
+        private void SetupDebugTargetInfoForWebkitV2Protocol(ref VsDebugTargetInfo vsDebugTargetInfo, DebugLaunchActionContext debugLaunchContext, string nodeExe)
+        {
+            // todo: refactor the debugging and process starting so we can re-use
+
+            var setupConfiguration = new SetupConfiguration();
+            var setupInstance = setupConfiguration.GetInstanceForCurrentProcess();
+            var visualStudioInstallationInstanceID = setupInstance.GetInstanceId();
+
+            // The Node2Adapter depends on features only in Node v6+, so the old v5.4 version of node will not suffice for this scenario
+            // This node.exe will be the one used by the node2 debug adapter, not the one used to host the user code.
+            var pathToNodeExe = Path.Combine(setupInstance.GetInstallationPath(), "JavaScript\\Node.JS\\v6.4.0_x86\\Node.exe");
+
+            // We check the registry to see if any parameters for the node.exe invocation have been specified (like "--inspect"), and append them if we find them.
+            var nodeParams = NodejsProjectLauncher.CheckForRegistrySpecifiedNodeParams();
+            if (!string.IsNullOrEmpty(nodeParams))
+            {
+                pathToNodeExe = pathToNodeExe + " " + nodeParams;
+            }
+
+            var pathToNode2DebugAdapterRuntime = Environment.ExpandEnvironmentVariables(@"""%ALLUSERSPROFILE%\" +
+                    $@"Microsoft\VisualStudio\NodeAdapter\{visualStudioInstallationInstanceID}\extension\out\src\nodeDebug.js""");
+
+            string trimmedPathToNode2DebugAdapter = pathToNode2DebugAdapterRuntime.Replace("\"", "");
+            if (!File.Exists(trimmedPathToNode2DebugAdapter))
+            {
+                pathToNode2DebugAdapterRuntime = Environment.ExpandEnvironmentVariables(@"""%ALLUSERSPROFILE%\" +
+                    $@"Microsoft\VisualStudio\NodeAdapter\{visualStudioInstallationInstanceID}\out\src\nodeDebug.js""");
+            }
+
+            var target = vsDebugTargetInfo.bstrExe;
+            var cwd = Path.GetDirectoryName(target); // Current working directory
```

Comment:
Yes, this is the open folder scenario though. We don't use the project properties there.